### PR TITLE
Cleanup the Admission Webhook E2E code

### PR DESCRIPTION
**Pull request by jasonrichardsmith**
**What this PR does / why we need it**:
The current documentation for Admission Webhooks points to the e2e tests as an example.
The code appears to have some legacy configuration left behind.  The variables are confusingly named and it lacked comments developers could follow.  This is an attempt to clean up some code and make the code more readable to make webhook development easier.

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

File: test/images/webhook/crd.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	"fmt"
+
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/golang/glog"
+	"k8s.io/api/admission/v1beta1"
+)
+
+func mutateCustomResource(ar v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
```
Comment (caesarxuchao) [inline test/images/webhook/crd.go]: I would put CR and CRD separately.

File: test/images/webhook/main.go [hunk 2]
```diff
@@ -309,31 +61,32 @@ func serve(w http.ResponseWriter, r *http.Request, admit admitFunc) {
...
-		response.Response = reviewResponse
-		response.Response.UID = ar.Request.UID
-	}
-	// reset the Object and OldObject, they are not needed in a response.
-	ar.Request.Object = runtime.RawExtension{}
-	ar.Request.OldObject = runtime.RawExtension{}
+	glog.V(2).Info(fmt.Sprintf("sending response: %v", responseAdmissionReview.Response))
+
+	// Return the same UID
+	responseAdmissionReview.Response.UID = requestedAdmissionReview.Request.UID
```
Comment (caesarxuchao) [inline test/images/webhook/main.go]: Move this line to before the glog.V(2).Info.

File: test/images/webhook/main.go [hunk 3]
```diff
@@ -309,31 +61,32 @@ func serve(w http.ResponseWriter, r *http.Request, admit admitFunc) {
...
 
-	resp, err := json.Marshal(response)
+	respbytes, err := json.Marshal(responseAdmissionReview)
```
Comment (caesarxuchao) [inline test/images/webhook/main.go]: nit: s/respbytes/respBytes

File: test/images/webhook/patches.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+const (
+	patch1 string = `[
+         { "op": "add", "path": "/data/mutation-stage-1", "value": "yes" }
+     ]`
+	patch2 string = `[
+         { "op": "add", "path": "/data/mutation-stage-2", "value": "yes" }
+     ]`
+	addInitContainerPatch string = `[
+		 {"op":"add","path":"/spec/initContainers","value":[{"image":"webhook-added-image","name":"webhook-added-init-container","resources":{}}]}
+	]`
```
Comment (caesarxuchao) [inline test/images/webhook/patches.go]: Could you move these patches to the handlers that use them? You can duplicate them if they are used by multiple handlers.

**Review by caesarxuchao — COMMENTED** [review]
A couple of nits and otherwise lgtm. I'll send a followup to update the version tag in the Makefile and e2e tests to actually put the new code in use.

**Comment by jasonrichardsmith** [comment]
@caesarxuchao I think I made all the expected changes, and made the patch_tests.go work a little better.

**Comment by caesarxuchao** [comment]
@jasonrichardsmith sorry I missed this thread. Please ping me on Slack directly if i'm not responding on github.

**Comment by jasonrichardsmith** [comment]
I updated the build file to include all the missing go packages.

File: test/images/webhook/alwaysdeny.go [hunk 5]
```diff
@@ -1,3 +1,19 @@
+/*
+Copyright 2017 The Kubernetes Authors.
```
Comment (yue9944882) [inline test/images/webhook/alwaysdeny.go]: 2018?

Comment (jasonrichardsmith) [reply test/images/webhook/alwaysdeny.go]: Will fix

**Comment by caesarxuchao** [comment]
Could you update the TAG in https://github.com/kubernetes/kubernetes/blob/master/test/images/webhook/Makefile#L16 to 1.11v1,

and update the tag here: https://github.com/kubernetes/kubernetes/blob/master/test/utils/image/manifest.go#L51

I'll build & push the image after you make the change, and then we'll merge if CI still passes.

**Comment by caesarxuchao** [comment]
I pushed the image and let's wait for the test result.

**Comment by caesarxuchao** [comment]
Actually it should be 1.12v1, but let's wait for the test results first.

**Comment by jasonrichardsmith** [comment]
I saw tests passed so I updated the image number.

**Comment by caesarxuchao** [comment]
Pushed the new images. Let's see the test result. Please squash before merging.

**Comment by caesarxuchao** [comment]
lgtm. Please squash. Thanks.

**Comment by caesarxuchao** [comment]
/assign @luxas
for approval. Thanks.

**Comment by caesarxuchao** [comment]
/assign @ixdy 
for approval. Thanks.

**Comment by caesarxuchao** [comment]
@jasonrichardsmith  unfortunately it needs yet another rebase.

@ixdy @luxas @mkumatag could you help approve the PR? Thank you.

**Comment by caesarxuchao** [comment]
/assign @mkumatag
for approval, thanks.

**Comment by ixdy** [comment]
/approve
though this needs rebase

**Pull request merged**